src/server_manager/model/gcp.ts

```diff
@@ -31,6 +36,61 @@ export type BillingAccount = {
   name: string,
 };
 
+export type CreationParams = {
+  cloudProvider: CloudProvider.GCP;
+  /** The GCP project ID. */
+  projectId: string;
+  /** The ID of the GCP zone to create the server in. */
+  zoneId: string;
+};
+
+/**
+ * Represents a Cloud region as a hierarchy of place names,
+ * e.g. city, state, country, continent.
+ */
+export class Region {
+  constructor(private divisions: string[]) {}
+
+  getFullName(): string {
+    return this.divisions.join(', ');
+  }
+
+  getFirstName(): string {
+    return this.divisions[0];
+  }
+}
+
+// TODO: Map regions to country codes.
+// TODO: Localize place names.
+/** @see https://cloud.google.com/compute/docs/regions-zones */
+export const LOCATION_MAP: {[regionId: string]: Region} = {
```

Comment:
Let's make this map be GcpRegionId -> city, region, country ids.
We will need the country codes for the flags.